examples/templates/docker/main.tf

```diff
@@ -80,11 +80,7 @@ resource "docker_container" "workspace" {
   dns      = ["1.1.1.1"]
   # Use the docker gateway if the access URL is 127.0.0.1
   command = [
-    "sh", "-c",
-    <<EOT
-    trap '[ $? -ne 0 ] && echo === Agent script exited with non-zero code. Sleeping infinitely to preserve logs... && sleep infinity' EXIT
-    ${replace(coder_agent.main.init_script, "/localhost|127\\.0\\.0\\.1/", "host.docker.internal")}
-    EOT
+    "sh", "-c", replace(coder_agent.main.init_script, "/localhost|127\\.0\\.0\\.1/", "host.docker.internal")
```

Comment:
This may make debugging harder since the container will get immediately deleted AFAIK

Reply:
We already have a trap in the bootstrap script, I think this is a remnant from a past implementation before that change.

This one also uses `sleep infinity` which isn’t compatible with all docker images.